crates/ruff_linter/src/rules/flake8_datetimez/rules/call_datetime_strptime_without_zone.rs

```diff
@@ -102,11 +102,23 @@ pub(crate) fn call_datetime_strptime_without_zone(checker: &mut Checker, call: &
     }
 
     // Does the `strptime` call contain a format string with a timezone specifier?
-    if let Some(Expr::StringLiteral(ast::ExprStringLiteral { value: format, .. })) =
-        call.arguments.args.get(1).as_ref()
-    {
-        if format.to_str().contains("%z") {
-            return;
+    if let Some(expr) = call.arguments.args.get(1) {
+        match expr {
+            Expr::StringLiteral(ast::ExprStringLiteral { value, .. }) => {
+                if value.to_str().contains("%z") {
+                    return;
+                }
+            }
+            Expr::FString(ast::ExprFString { value, .. }) => {
+                // TODO(dhruvmanila): This doesn't consider f-strings that are implicitly concatenated
+                // to strings. For example, `f"%Y-%m-%dT%H:%M:%S{('.%f' if millis else '')}" "%z"`
```

Comment:
Should we check in a test case for this anyway?

Reply:
Ruff will raise a violation in this case even though it shouldn't. Maybe checking string literal isn't too bad 🤔

Reply:
I think it's find to have an "incorrect" test case as long as it's clearly documented, then we have a test demonstrating the fix when you get to it.

Reply:
I see. I've added a commit which also checks for this case.